### PR TITLE
Add server ready check before starting evaluation

### PR DESCRIPTION
**Pull request by athitten**
# What does this PR do ?

Adds method `wait_for_server_ready` to `evaluation/base.py` that waits for the PyTriton server to be ready and the specific model (with name defined in `model_name` arg) to be loaded and ready before starting evaluation.

Adds bool [generation_logits_available](https://github.com/NVIDIA/NeMo/pull/11731/files#diff-3a3bd49eed8a038989253f6f2e0a01cf494d1de391863d4ab2377a1f7d4cf832R1152) to `triton_infer_fn` since check via `if "output_generation_logits" in inputs` is not robust in cases where `output_generation_logits` is popped from inputs dict in the previous line.

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: nemo/collections/llm/api.py [hunk 1]
```diff
@@ -329,7 +329,6 @@ def deploy(
     triton_http_port: int = 8000,
     triton_grpc_port: int = 8001,
     triton_http_address: str = "0.0.0.0",
-    triton_request_timeout: int = 60,
     triton_model_repository: Path = None,
```
Comment (athitten) [inline nemo/collections/llm/api.py:332]: `triton_request_timeout` unused hence removed.

**Review by HuiyingLi — APPROVED** [review]
LGTM!
